File: app/src/main/java/localhost/toolkit/text/ErrorEqualsListener.java
package localhost.toolkit.text;

import android.widget.EditText;

public class ErrorEqualsListener extends AbstractErrorListener {
    private EditText reference;

    public ErrorEqualsListener(String errorMsg, EditText subject, EditText reference) {
        super(errorMsg, subject);
        this.reference = reference;
    }

    @Override
    public boolean matches() {
        return editText.getText().toString().equals(reference.getText().toString());
    }
}

File: core/src/main/kotlin/it/localhostsoftware/core/text/ErrorEqualsListener.kt
package it.localhostsoftware.core.text

import android.widget.EditText

class ErrorEqualsListener(errorMsg: String?, subject: EditText, private val reference: EditText) : AbstractErrorListener(errorMsg, subject) {
    override fun matches() =
        editText.text.toString() == reference.text.toString()
}